app/services/search/base.rb

```diff
@@ -2,39 +2,72 @@ module Search
   class Base
     class << self
       def index(doc_id, serialized_data)
-        SearchClient.index(
-          id: doc_id,
-          index: self::INDEX_ALIAS,
-          body: serialized_data,
-        )
+        request do
```

Comment:
I am not a fan of having to wrap every single Elasticsearch request we make in this request block. I think it will be easy to forget and become cumbersome. I would rather make a wrapper for the search client that we call everywhere and implement something like this
```ruby
class ElasticsearchClientWrapper #pick a better name
  attr_accessor :connection

  def initialize(options = {})
    @connection = Elasticsearch::Client.new
  end

  def method_missing(m, *args, &block)
    @connection.send(m, *args, &block)
  rescue Elasticsearch::Transport::Transport::Error => e
    SearchErrorHandler.new(e).report(m, args) # This could then handle logging, raising, Datadog, Honeybadger, etc
  end
end
```
Then we replace `SearchClient` with this wrapper everywhere and that becomes our search client that we always work through.